sw/device/lib/dif/README.md

```diff
@@ -81,6 +81,31 @@ Notational caveats:
 * Unless otherwise noted, all symbols mentioned below are required.
 
 
+#### Hardware Parameterization
+
+Our aim is that a single DIF library can be used with multiple instances of the
+same IP on the same chip, even when those IPs have been instantiated with
+different hardware parameters.
+
+At the moment, we have a good approach to being able to address separate
+hardware instances instantiated at separate addresses, as long as they have the
+same hardware parameters (see the `base_addr` member in `dif_<ip>_params_t`).
+Most other parameters come from the specific IP on a case-by-case basis.
+
+As much as possible, we would like to hide these parameters underneath the
+interface of the DIF, but this is not always possible, especially where
+particular functionality requires the DIF caller to allocate memory.
+
+In order to support exposing these parameters to callers, DIFs should provide
+query functions which take a `dif_<ip>_params_t`, rather than `#define`s or
+global variable definitions. This should be done even if the DIF does not
+support parameters, so we can add this parameterization later.
```

Comment:
I'd consider updating the DIF template as part of this change, so people start adding them ASAP.